lib/src/zenohRpcClient.cpp

```diff
@@ -112,121 +117,215 @@ UStatus ZenohRpcClient::term() noexcept {
+    z_owned_reply_channel_t *channel = nullptr;
+    UStatus status;
 
-    opts.timeout_ms = requestTimeoutMs_;
-    opts.attachment = z_bytes_map_as_attachment(&map);
+    do {
 
-    if ((0 != payload.size()) && (nullptr != payload.data())) {
-        opts.value.payload.len =  payload.size();
-        opts.value.payload.start = payload.data();
-    } else {
-        opts.value.payload.len = 0;
-        opts.value.payload.start = nullptr;
-    }
+        status.set_code(UCode::INTERNAL);
 
-    z_bytes_map_insert_by_alias(&map, z_bytes_new("header"), bytes);
+        auto uriHash = std::hash<std::string>{}(LongUriSerializer::serialize(topic));
+        auto uuid = Uuidv8Factory::create();
+    
+        UAttributesBuilder builder(uuid, UMessageType::UMESSAGE_TYPE_REQUEST, UPriority::UPRIORITY_CS0);
 
-    if (0 != z_get(z_loan(session_), z_keyexpr(std::to_string(uriHash).c_str()), "", z_move(channel->send), &opts)) {
-        spdlog::error("z_get failure");
-        z_drop(&map);
-        return future;
-    }
+        if (options.has_ttl()) {
+            builder.setTTL(options.ttl());
+            opts.timeout_ms = options.ttl();
+        } else {
+            opts.timeout_ms = requestTimeoutMs_;
+        }
+
+        builder.setPriority(options.priority());
+
+        UAttributes attributes = builder.build();
 
-    future = threadPool_->submit(handleReply, channel);
+        // Serialize UAttributes
+        size_t attrSize = attributes.ByteSizeLong();
+        std::vector<uint8_t> serializedAttributes(attrSize);
+        if (!attributes.SerializeToArray(serializedAttributes.data(), attrSize)) {
+            spdlog::error("attributes SerializeToArray failure");
+            break;
+        }
+
+        z_bytes_t bytes = {.len = serializedAttributes.size(), .start = serializedAttributes.data()};
+        
+        z_bytes_map_insert_by_alias(&map, z_bytes_new("attributes"), bytes);
+
+        channel = new z_owned_reply_channel_t;
+        if (nullptr == channel) {
+            spdlog::error("failed to allocate channel");
+            break;
+        }
+
+        *channel = zc_reply_fifo_new(16);
+
+        opts.attachment = z_bytes_map_as_attachment(&map);
+
+        if ((0 != payload.size()) && (nullptr != payload.data())) {
+            opts.value.payload.len =  payload.size();
+            opts.value.payload.start = payload.data();
+        } else {
+            opts.value.payload.len = 0;
+            opts.value.payload.start = nullptr;
+        }
+        
+        if (0 != z_get(z_loan(session_), z_keyexpr(std::to_string(uriHash).c_str()), "", z_move(channel->send), &opts)) {
+            spdlog::error("z_get failure");
+            break;
+        }
+       
+        future = threadPool_->submit([=] { return handleReply(channel, callback); });
+        if (false == future.valid()) {
+            spdlog::error("invalid future received");
+            break;
+        }
+    
+        status.set_code(UCode::OK);
+
+    } while(0);
```

Comment:
I'll admit - I don't exactly understand what is going on here in this loop. It looks like there's no exit from the loop other than through an error, so if it succeeds in invoking the method and getting a future from the thread pool, it will then... go back to the start of the loop and try again?

Another concern: If we iterate through this loop multiple times, every iteration calls `new` on line 223 without a corresponding `delete` (the only `delete` is outside the loop). **This would result in a memory leak** - it would probably be safer to use smart pointers (e.g. `std::unique_ptr` and `std::make_unique`) instead of bare pointers.

Reply:
But why would i want to allocate memory during run time when i dont actually need it?

Reply:
Where would memory be allocated without needing it? Simply changing `z_owned_reply_channel_t *channel` to `std::unique_ptr<z_owned_reply_channel_t> channel` doesn't change your memory allocation pattern.